### PR TITLE
MM-13260: Adding AdminPanel component to be used in the admin console

### PR DESCRIPTION
**Pull request by jespino**
#### Summary
Adding AdminPanel component to be used in the admin console

#### Ticket Link
[MM-13260](https://mattermost.atlassian.net/browse/MM-13260)

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/admin_console/permission_schemes_settings/permission_schemes_settings.jsx [hunk 1]
```diff
@@ -159,72 +159,28 @@ export default class PermissionSchemesSettings extends React.PureComponent {
...
+                <AdminPanelWithButton
+                    titleId={t('admin.permissions.systemSchemeBannerTitle')}
+                    titleDefault='System Scheme'
+                    subtitleId={t('admin.permissions.systemSchemeBannerText')}
+                    subtitleDefault='Set the default permissions inherited by all teams unless a [Team Override Scheme](!https://about.mattermost.com/default-team-override-scheme) is applied.'
+                    url='/admin_console/permissions/system-scheme'
+                    disabled={teamOverrideView !== null}
+                    buttonTextId={t('admin.permissions.systemSchemeBannerButton')}
+                    buttonTextDefault='Edit Scheme'
+                />
```
Comment (mkraft) [inline components/admin_console/permission_schemes_settings/permission_schemes_settings.jsx]: I like this API you have made 👍

File: components/admin_console/permission_schemes_settings/permission_schemes_settings.jsx [hunk 2]
```diff
@@ -250,7 +206,7 @@ export default class PermissionSchemesSettings extends React.PureComponent {
                                 />
                             </LoadingWrapper>
                         </button>}
-                </div>
```
Comment (mkraft) [inline components/admin_console/permission_schemes_settings/permission_schemes_settings.jsx]: And I really like the use of children 👍

File: components/admin_console/permission_schemes_settings/permission_system_scheme_settings/permission_system_scheme_settings.jsx [hunk 3]
```diff
@@ -216,136 +217,78 @@ export default class PermissionSystemSchemeSettings extends React.Component {
...
+
+                <AdminPanelTogglable
+                    className='permissions-block'
+                    open={this.state.openRoles.channel_admin}
+                    onToggle={() => this.toggleRole('channel_admin')}
+                    titleId={t('admin.permissions.systemScheme.channelAdminsTitle')}
+                    titleDefault='Channel Administrators'
+                    subtitleId={t('admin.permissions.systemScheme.channelAdminsDescription')}
+                    subtitleDefault='Permissions granted to channel creators and any users promoted to Channel Administrator.'
+                >
```
Comment (mkraft) [inline components/admin_console/permission_schemes_settings/permission_system_scheme_settings/permission_system_scheme_settings.jsx]: And these 👍

File: components/widgets/admin_console/admin_panel_with_button.jsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            className={'AdminPanelWithButton ' + props.className}
+            id={props.id}
+            titleId={props.titleId}
+            titleDefault={props.titleDefault}
+            subtitleId={props.subtitleId}
+            subtitleDefault={props.subtitleDefault}
+            button={button}
+        >
+            {props.children}
+        </AdminPanel>
```
Comment (mkraft) [inline components/widgets/admin_console/admin_panel_with_button.jsx]: Love that this uses `<AdminPanel>`.

**Comment by jespino** [comment]
@mkraft Thanks!, I sitll have to use this component in the LDAP Groups admin configuration, so, I'll do it and ask you to "re-review" that part.

**Comment by jespino** [comment]
@mkraft PTAL, you only need to review the last commit.

File: components/admin_console/group_settings/__snapshots__/group_settings.test.jsx.snap [hunk 5]
```diff
@@ -27,10 +27,11 @@ For more information on Groups, please see [documentation](!https://about.matter
     </div>
   </div>
   <AdminPanel
+    className=""
```
Comment (mkraft) [inline components/admin_console/group_settings/__snapshots__/group_settings.test.jsx.snap]: Does it make sense for this property to be required if it's often an empty string?

Comment (jespino) [reply components/admin_console/group_settings/__snapshots__/group_settings.test.jsx.snap]: Yes, I don't have an strong opinion here, it's just easier to just have a isRequiered + default than handle it inside. This way, if you don't pass it I don't need to handle the null case.

**Comment by jespino** [comment]
Rebased and waiting for @grundleborg's review

File: components/widgets/admin_console/admin_panel.jsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            }
+        </div>
+        {props.children}
+    </div>
+);
+
+AdminPanel.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string.isRequired,
+    id: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (jespino) [reply components/widgets/admin_console/admin_panel.jsx]: Exactly, for E2E tests.

File: components/widgets/admin_console/admin_panel.jsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    titleId: PropTypes.string.isRequired,
+    titleDefault: PropTypes.string.isRequired,
+    subtitleId: PropTypes.string.isRequired,
+    subtitleDefault: PropTypes.string.isRequired,
+    onHeaderClick: PropTypes.func,
```
[the remark this answers is not in the document]
Comment (jespino) [reply components/widgets/admin_console/admin_panel.jsx]: I mix them all the time, the "standard" used by the react I think is `onClick`, `onChange`, `onBlur`, `onFocus`, etcetera, probably would be a good policy to follow that approach.

File: components/widgets/admin_console/admin_panel_togglable.jsx [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+AdminPanelTogglable.propTypes = {
+    children: PropTypes.node,
+    className: PropTypes.string.isRequired,
+    id: PropTypes.string,
+    open: PropTypes.bool.isRequired,
+    titleId: PropTypes.string.isRequired,
+    titleDefault: PropTypes.string.isRequired,
+    subtitleId: PropTypes.string.isRequired,
+    subtitleDefault: PropTypes.string.isRequired,
+    onToggle: PropTypes.func.isRequired,
```
[the remark this answers is not in the document]
Comment (jespino) [reply components/widgets/admin_console/admin_panel_togglable.jsx]: Yes, I think we definitively need to have that in the style guide, a decision about use it as `onClick` or `onClicked`.

File: components/widgets/admin_console/admin_panel_with_button.jsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+import {Link} from 'react-router-dom';
+
+import AdminPanel from './admin_panel.jsx';
+
+const AdminPanelWithButton = (props) => {
```
[the remark this answers is not in the document]
Comment (jespino) [reply components/widgets/admin_console/admin_panel_with_button.jsx]: We could split that in `AdminPanelWithButton` and `AdminPanelWithLink`, I going to do this, and lets see how it ends.

[the remark this answers is not in the document]
Comment (jespino) [reply components/widgets/admin_console/admin_panel_with_button.jsx]: Splitted :)

**Pull request merged**